account_invoice_report_hls/report/invoice_delivery_report.py

```diff
@@ -64,7 +64,9 @@ class InvoiceDeliveryReportLine(models.TransientModel):
 
     def _create_invoice_delivery_report_lines(self, report, invoice_line_ids,
                                               date_from, date_to):
-        for il in invoice_line_ids.filtered(lambda x: x.product_id.type != 'service'):
+        for il in invoice_line_ids  \
+                    .filtered(lambda x: x.product_id.type != 'service')  \
+                    .sorted(key=lambda r: r.sale_line_ids.id):
```

Comment:
@nakometal Can you check if `r.sale_line_ids[0].order_id.id` would work?  There could be situations where sorting by `r.sale_line_ids.id` won't generate the expected result in case a line has been additionally added to an existing sales order, etc.

Reply:
@nakometal Or, simply use `r.sale_order_name` as you originally tried.  I now think going for this option is reasonable, to make the logic consistent with the sorting of account.invoice.line presentation.